### PR TITLE
Multi-platform builds of kubernetes/ingress-gce using github actions

### PR DESCRIPTION
**Pull request by kron4eg**
**What this PR does / why we need it**:
This enables building multiplatform image of the upstream kubernetes/ingress-gce.

**Which issue(s) this PR fixes**:
xref: https://github.com/gardener/gardener-extension-provider-gcp/issues/829

**Release note**:

```feature dependency
Multi-platform builds of kubernetes/ingress-gce
```

### CODE REVIEW

**Comment by ghost** [comment]
@kron4eg Thank you for your contribution.

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
+name: Build and push a release image
+on:
+  push:
+    tags:
+      - '**'
```
Comment (ccwienk) [inline .github/workflows/release.yaml]: So you plan to trigger build + (image-)push by pushing a (git-)tag? I would have expected to e.g. pass-in version via input + have the pipeline publish a tag (there might be cases where build will fail, which will require resetting tags, if I understand the approach correctly - which does not seem like a good practice to me) :-/

Comment (kron4eg) [reply .github/workflows/release.yaml]: Sounds good. I've re-done it using workflow_dispatch inputs that should be provided, and new release will be done by the GH action itself.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+#
+# SPDX-License-Identifier: Apache-2.0
+
+.ONESHELL:
+.SHELLFLAGS 	 := -eu -o pipefail -c
+SHELL 			 := bash
+MAKEFLAGS 		 += --warn-undefined-variables
+MAKEFLAGS 		 += --no-builtin-rules
+IMAGE_REPOSITORY := ghcr.io/gardener/ingress-gce
+IMAGE_TAG        := $(shell git describe --tags --always)
```
Comment (ScheererJ) [inline Makefile]: Could you please replace tab characters with spaces?

```suggestion
.SHELLFLAGS      := -eu -o pipefail -c
SHELL            := bash
MAKEFLAGS        += --warn-undefined-variables
MAKEFLAGS        += --no-builtin-rules
IMAGE_REPOSITORY := ghcr.io/gardener/ingress-gce
IMAGE_TAG        := $(shell git describe --tags --always)
```

File: README.md [hunk 3]
```diff
@@ -1,7 +1,46 @@
...
 
 [![reuse compliant](https://reuse.software/badge/reuse-compliant.svg)](https://reuse.software/)
 
+This repository is dedicated to building [kubernetes/ingress-gce](https://github.com/kubernetes/ingress-gce) images for Gardener project.
+
+## Building a new Release
+
+Once in a while a new release of the
+[kubernetes/ingress-gce](https://github.com/kubernetes/ingress-gce) will be
+released and we need to build new docker image. In order to trigger the new
```
Comment (ScheererJ) [inline README.md]:
```suggestion

## Building a new Release

Once in a while a new release of the
[kubernetes/ingress-gce](https://github.com/kubernetes/ingress-gce) will be
released and we need to build new container image. In order to trigger the new
```

**Review by ScheererJ — CHANGES_REQUESTED** [review]

File: Makefile (hunk 2 above)
Comment (kron4eg) [reply Makefile]: done

**Review by ScheererJ — APPROVED** [review]
/lgtm

**Pull request merged**